daemon/config: remove commonUnixBridgeConfig and CommonUnixConfig

thaJeztah opened the pull request:
These types were added as part of the "Solaris" support, because Solaris didn't support these options. Having these types added another layer of "abstraction" when reading the code, making the code harder to read, and more complicated to use.

Now that implementing support for Solaris has been abandoned (and removed), we can re-integrate these fields in the "unix" configuration.

See individual commits for details.

thaJeztah committed:
daemon/config: reorganize code between unix and windows files

Put variables and functions in the same owrder between both,
to allow for easier comparing between platforms.

Also synchronised some comments/godoc between both.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
daemon/config: remove commonUnixBridgeConfig

This type was added to support Solaris (which didn't support these
options). Solaris support was removed, so we can integrate this type
back into the "unix" type.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
daemon/config: remove CommonUnixConfig type

This type was added to support Solaris (which didn't support these
options). Solaris support was removed, so we can integrate this type
back into the "unix" type.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
daemon/config: move "common" tests

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
daemon/config.New(): rewrite to be slightly more idiomatic

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

AkihiroSuda commented on daemon/config/config_unix.go:
```diff
@@ -1,10 +1,12 @@
-// +build linux freebsd
+// +build !windows
```
Can we rename the file to _linux.go and remove the build tag?

I don’t think the code in this file works on non-Linux.

thaJeztah replied:
Yeah, makes sense; I thought "if it compiles, it compiles" but probably better to be explicit (as we for sure don't _test_, or _assume_ other unix-y environments for these)

Updated 👍

thaJeztah committed:
daemon/config: restrict "unix" code is linux

This code is not generically useful on "unix", and contains linux-
specific code, so make it only compile on linux.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented on daemon/config/config_linux.go:
```diff
@@ -104,7 +160,7 @@ func (conf *Config) ValidatePlatformConfig() error {
 }
 
-// IsRootless returns conf.Rootless
+// IsRootless returns conf.Rootless on Unix but false on Windows
```
Let me change this one as well

thaJeztah commented:
@tianon @cpuguy83 PTAL 🤗 (best reviewed by going through the commits)

tianon reviewed (approved):
This LGTM, although there has been some effort recently around FreeBSD -- I suppose these are split nicely enough to be easily reverted if that work happens to start landing before this code changes again. :sweat_smile:

thaJeztah commented:
>  I suppose these are split nicely enough to be easily reverted if that work happens to start landing before this code changes again.

Yes, I had that in the back of my mind when making these changes, then considered "YAGNI"; if we need to specialise these types again, it will _probably_ be slightly different, and I thought it'd be better to make changes based on what we need _then_ instead of hoping it'll be the same as it is now.

thaJeztah commented:
Let me get this one in
